### ISSUE TITLE
Mutable Data from runTransactionBlock cannot be cast

### ISSUE THREAD

**Issue by marknf94** [firebase/firebase-ios-sdk#989]
### [READ] Step 1: Are you in the right place?

Yes!

### [REQUIRED] Step 2: Describe your environment

  * Xcode version: 9.2
  * Firebase SDK version: 4.6
  * Firebase Component:  Database
  * Component version: 4.1.1

### [REQUIRED] Step 3: Describe the problem

When running a transaction block after getting the current data, I try to cast it to a [String: Any]. In my debugger the if statement always fails during the cast, causing me not to be able to get the dictionary to parse the data needed. 

`

          let ogPostReference = ref.child("posts").child("-L8WBkVp9Yd5JmGd10lH")
           ogPostReference.runTransactionBlock({ (currentData: MutableData) -> TransactionResult in

                if var post = currentData.value as? [String: Any]  { // Fails here

                    var count: Int
                    count = post["commentCount"] as? Int ?? 0
                    count += 1
                    post["count"] = stars as AnyObject?
                     currentData.value = post
                    
                    return TransactionResult.success(withValue: currentData)
                }
                return TransactionResult.success(withValue: currentData)
            }) { (error, committed, snapshot) in
                if let error = error {
                    print(error.localizedDescription)
                }
            }`

My structure on fire base looks like this 

<img width="418" alt="screen shot 2018-03-28 at 1 24 17 am" src="https://user-images.githubusercontent.com/12654342/38017403-c930fd2a-3226-11e8-8da5-6afb918a37c0.png">

**Comment by morganchen12** [comment]
What's the value of `currentData.value`? Based on your console screenshot, it looks like it should be of type `[Any]`, not `[String: Any]`.

**Comment by marknf94** [comment]
Can't seem to get the data to show in the console, seems like it might be null? Maybe my reference path is incorrect? 
![screen shot 2018-03-28 at 11 15 34 am](https://user-images.githubusercontent.com/12654342/38048173-83c3e68c-3279-11e8-8c38-02e888462cc1.png)
![screen shot 2018-03-28 at 11 15 42 am](https://user-images.githubusercontent.com/12654342/38048183-88b2d342-3279-11e8-943f-d801107f4951.png)

**Comment by morganchen12** [comment]
I misread your original comment, sorry.

Are you able to fetch data from that location outside of a transaction, or is it always nil?

**Comment by marknf94** [comment]
Seems like the issue with the casting was with my runTransactionBlock being inside of a updateChildValues call. (I wanted to only run the  transaction if the updateChildValues block success = true). I tried to run a oberveSingleEvent of in the same place as per @morganchen12 comment to see if I can get the data outside of a transaction. It was the same issue of not being to cast it. Once I moved it outside of the updateChildValues call, they both started working.
